pykern/pkunit.py

```diff
@@ -555,6 +560,30 @@ def _failed_msg(process):
 """
             )
 
+        def _ndiff_config(epsilon, work_d):
+            pykern.pkio.write_text(
+                work_d.join(_NDIFF_CONF_FILE), f"* * abs={float(epsilon)}"
+            )
+
+        def _ndiff_files(expect_path, actual_path, epsilon):
+            _ndiff_config(epsilon, w)
+            p = subprocess.run(
+                (
+                    "ndiff",
+                    actual_path,
+                    expect_path,
+                    w.join(_NDIFF_CONF_FILE),
+                ),
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+            )
+            d = pkcompat.from_bytes(p.stderr)
```

Comment:
Test the process status on `p` if ndiff does return a status (like `diff` does). Testing for a specific string isn't as robust especially if ndiff crashes for a different reason.

Reply:
So ndiff wont error if it finds diff (no fail fast just writes to file). Should I do something like
```python
if p.returncode != 0:
    pkfail("ndiff failed: {}", p.stderr)
```
just in case ndiff fails for some other reason and we want to know more?

Reply:
The failure cases outnumber the success cases so it is more robust then to test for the success case. This makes the code more robust. What does it output on success?

Reply:
on success it will just output how many lines have been diffed eg.
```
3 lines have been diffed
```

Reply:
So you would test for exactly `^\d+ lines have been diffed\n*$`. The goal is to know what success is without getting collisions on possible failures.

Reply:
Both failure and success will actually say 
```
n lines have been diffed
```
and it is failures that result in 
```
n diffs detected
```

So in that case would it make more sense to test p.stderr for `n diffs detected` ?